web/client/components/map/openlayers/Map.jsx

```diff
@@ -190,18 +200,50 @@ class OpenlayersMap extends React.Component {
         }
 
         if (this.map && (this.props.projection !== newProps.projection || this.haveResolutionsChanged(newProps))) {
+            let mapProjection = newProps.projection;
             const center = CoordinatesUtils.reproject([
                 newProps.center.x,
                 newProps.center.y
-            ], 'EPSG:4326', newProps.projection);
+            ], 'EPSG:4326', mapProjection);
             this.map.setView(this.createView(center, newProps.zoom, newProps.projection, newProps.mapOptions && newProps.mapOptions.view));
+            const mapExtent = mapProjection && CoordinatesUtils.reprojectBbox(newProps.maxExtent, mapProjection, 'EPSG:4326');
+            // perform a check if the data and the projection are compatible
+            if (newProps.children) {
```

Comment:
Don't we have a similar check in the layer?

Reply:
this check is performed when the user changes the crs, the other one is performed when the user loads a new layer to the map

Reply:
ok